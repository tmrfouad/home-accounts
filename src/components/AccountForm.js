import React from 'react';
import { connect } from 'react-redux';
import {
  startAddAccount,
  startEditAccount,
  startRemoveAccount
} from '../actions/accounts';
import ConfirmModal from './ConfirmModal';

export class AccountForm extends React.Component {
  constructor(props) {
    super(props);
    const account = this.props.account;
    this.state = {
      name: account ? account.name : '',
      error: '',
      mode: account ? 'edit' : 'add',
      isModalOpen: false
    };
  }

  onNameChange = e => {
    this.setState({ name: e.target.value });
  };

  onFormSubmit = e => {
    e.preventDefault();
    if (!this.state.name) {
      this.setState({ error: 'Please enter a name!' });
    } else {
      if (this.state.mode === 'edit') {
        this.props.startEditAccount(this.props.account.id, {
          name: this.state.name
        });
      } else {
        if (this.props.accounts.find(acc => acc.name === this.state.name)) {
          this.setState({ error: 'This account already exists!' });
        } else {
          this.props.startAddAccount({
            name: this.state.name
          });
        }
      }
      this.props.history.push('/accounts');
    }
  };

  removeAccount = () => {
    this.props.startRemoveAccount(this.props.account.id);
    this.props.history.push('/accounts');
  };

  openRemoveItemDialog = e => {
    e.preventDefault();
    this.setState(() => ({ isModalOpen: true }));
  };

  closeRemoveItemDialog = () => {
    this.setState(() => ({ isModalOpen: false }));
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h2 className="page-header__title">
              {this.state.mode === 'edit' ? 'Edit' : 'Add'} Account
            </h2>
          </div>
        </div>
        <form className="form content-container" onSubmit={this.onFormSubmit}>
          {this.state.error && (
            <div className="form__error">{this.state.error}</div>
          )}
          <input
            type="text"
            className="text-input"
            value={this.state.name}
            onChange={this.onNameChange}
          />
          <div>
            <button className="button form__action">Save Account</button>
            <button
              className="button form__action button--secondary"
              onClick={this.openRemoveItemDialog}
            >
              Remove Account
            </button>
          </div>
        </form>
        <ConfirmModal
          id="confirmModal"
          messageTitle="Remove Account!"
          messageBody="Are you sure you want to remove this item?"
          isOpen={this.state.isModalOpen}
          onModalClose={this.closeRemoveItemDialog}
          onModalOk={this.removeAccount}
        />
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  account: state.accounts.find(t => t.id === props.match.params.id),
  accounts: state.accounts
});

const mapDispatchToProps = dispatch => ({
  startAddAccount: account => dispatch(startAddAccount(account)),
  startEditAccount: (id, account) =>
    dispatch(startEditAccount({ id }, account)),
  startRemoveAccount: id => dispatch(startRemoveAccount({ id }))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AccountForm);
