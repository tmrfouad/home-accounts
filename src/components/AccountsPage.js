import React from 'react';
import AccountList from './AccountList';

const AccountsPage = () => (
  <div>
    <AccountList />
  </div>
);

export default AccountsPage;
