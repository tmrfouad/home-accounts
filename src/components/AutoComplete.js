import React from 'react';

export default class AutoComplete extends React.Component {
  constructor(props) {
    super(props);

    if (
      !props.id ||
      !props.source ||
      !props.displayField ||
      !props.valueField ||
      !props.onChange
    ) {
      throw new Error(
        'You must supply the id, src, displayField, valueField and onChange properties for the AutoComplete component.'
      );
    }
  }

  updateInputs() {
    const selectedItem =
      this.props.source.find(
        itm => itm[this.props.valueField] + '' === this.props.value + ''
      ) || {};

    const textInput = document.getElementById(this.props.id + '-text');
    const hiddenInput = document.getElementById(this.props.id + '-hidden');

    textInput.value = selectedItem[this.props.displayField] || '';
    hiddenInput.value = selectedItem[this.props.valueField] || '';
  }

  componentDidUpdate() {
    this.updateInputs();
  }

  componentDidMount() {
    this.updateInputs();
    this.autocomplete();
  }

  autocomplete() {
    const self = this;
    const inp = document.getElementById(self.props.id + '-text');
    const src = self.props.source;
    const displayField = self.props.displayField;
    const valueField = self.props.valueField;
    const onChange = self.props.onChange;

    let currentFocus;

    inp.addEventListener('click', function() {
      showList(true);
    });

    inp.addEventListener('blur', function(e) {
      if (e.target.closest('.autocomplete')) {
        return;
      }
      let inputValue = inp.value;
      const foundObj = src.find(o => o[displayField] + '' === inputValue + '');
      if (foundObj) {
        const value = foundObj[valueField];
        onChange(value);
        closeAllLists();
      } else {
        if (isListOpen()) {
          const firstListItem = inp.parentNode.querySelector(
            '.autocomplete-active'
          );
          if (firstListItem) {
            const value = firstListItem.getElementsByTagName('input')[0].value;
            const firstObj = src.find(
              itm => itm[valueField] + '' === value + ''
            );
            onChange(value);
            closeAllLists();
          }
        }
      }
    });

    /*execute a function when someone writes in the text field:*/
    inp.addEventListener('input', function(e) {
      showList();
    });
    /*execute a function presses a key on the keyboard:*/
    inp.addEventListener('keydown', function(e) {
      let x = document.getElementById(this.id + 'autocomplete-list');
      if (x) x = x.getElementsByTagName('div');
      if (e.keyCode == 40) {
        if (isListOpen()) {
          /*If the arrow DOWN key is pressed,
          increase the currentFocus variable:*/
          currentFocus++;
          /*and and make the current item more visible:*/
          addActive(x);
        } else {
          showList(true);
        }
      } else if (e.keyCode == 38) {
        //up
        /*If the arrow UP key is pressed,
        decrease the currentFocus variable:*/
        currentFocus--;
        /*and and make the current item more visible:*/
        addActive(x);
      } else if (e.keyCode == 13) {
        /*If the ENTER key is pressed, prevent the form from being submitted,*/
        e.preventDefault();
        if (currentFocus > -1) {
          /*and simulate a click on the "active" item:*/
          if (x) x[currentFocus].click();
        }
      } else if (e.keyCode == 27) {
        closeAllLists();
      }
    });
    function showList(showAll) {
      if (showAll && isListOpen()) {
        closeAllLists();
        return;
      }

      let a,
        b,
        i,
        inputValue = inp.value;

      /*close any already open lists of autocompleted values*/
      closeAllLists();

      if (!showAll && !inputValue) {
        return false;
      }
      currentFocus = -1;
      /*create a DIV element that will contain the items (values):*/
      a = document.createElement('DIV');
      a.setAttribute('id', inp.id + 'autocomplete-list');
      a.setAttribute('class', 'autocomplete-items');
      /*append the DIV element as a child of the autocomplete container:*/
      inp.parentNode.appendChild(a);
      /*for each item in the array...*/
      for (i = 0; i < src.length; i++) {
        /*check if the item starts with the same letters as the text field value:*/
        const srcObj = src[i];
        const srcValue = srcObj[valueField];
        const srcDisplay = srcObj[displayField];
        if (
          srcDisplay.toUpperCase().includes(inputValue.toUpperCase()) ||
          showAll
        ) {
          /*create a DIV element for each matching element:*/
          b = document.createElement('DIV');
          if (showAll) {
            if (srcDisplay.toUpperCase() === inputValue.toUpperCase()) {
              b.innerHTML = '<strong>' + srcDisplay + '</strong>';
            } else {
              b.innerHTML = srcDisplay;
            }
          } else {
            /*make the matching letters bold:*/
            const valIndex = srcDisplay
              .toUpperCase()
              .indexOf(inputValue.toUpperCase());
            b.innerHTML = srcDisplay.substr(0, valIndex);
            b.innerHTML +=
              '<strong>' +
              srcDisplay.substr(valIndex, inputValue.length) +
              '</strong>';
            b.innerHTML += srcDisplay.substr(valIndex + inputValue.length);
          }

          /*insert a input field that will hold the current array item's value:*/
          b.innerHTML += "<input type='hidden' value='" + srcValue + "'>";
          /*execute a function when someone clicks on the item value (DIV element):*/
          b.addEventListener('click', function(e) {
            /*insert the value for the autocomplete text field:*/
            const value = this.getElementsByTagName('input')[0].value;
            const selectedItem = self.props.source.find(
              itm => itm[valueField] + '' === value + ''
            );
            const display = selectedItem
              ? selectedItem[self.props.displayField]
              : '';

            // inp.value = display;
            // self.state.value[displayField] = display;

            // valInp.value = value;
            /*close the list of autocompleted values,
            (or any other open lists of autocompleted values:*/
            closeAllLists();
            onChange(value);
          });
          a.appendChild(b);
        }
      }
    }
    function addActive(x) {
      /*a function to classify an item as "active":*/
      if (!x) return false;
      /*start by removing the "active" class on all items:*/
      removeActive(x);
      if (currentFocus >= x.length) currentFocus = 0;
      if (currentFocus < 0) currentFocus = x.length - 1;
      /*add class "autocomplete-active":*/
      x[currentFocus].classList.add('autocomplete-active');
    }
    function removeActive(x) {
      /*a function to remove the "active" class from all autocomplete items:*/
      for (var i = 0; i < x.length; i++) {
        x[i].classList.remove('autocomplete-active');
      }
    }
    function closeAllLists(elmnt) {
      /*close all autocomplete lists in the document,
      except the one passed as an argument:*/
      let x = inp.parentNode.getElementsByClassName('autocomplete-items');
      for (var i = 0; i < x.length; i++) {
        if (elmnt != x[i] && elmnt != inp) {
          x[i].parentNode.removeChild(x[i]);
        }
      }
    }
    function isListOpen() {
      return !!inp.parentNode.querySelector('.autocomplete-items');
    }
    /*execute a function when someone clicks in the document:*/
    document.addEventListener('click', function(e) {
      closeAllLists(e.target);
    });
  }

  // <div className="backline" />

  render() {
    return (
      <div className="autocomplete">
        <input
          id={this.props.id + '-text'}
          type="text"
          className={this.props.className}
          placeholder={this.props.placeholder}
          autoComplete="off"
        />
        <input id={this.props.id + '-hidden'} type="hidden" />
      </div>
    );
  }
}
