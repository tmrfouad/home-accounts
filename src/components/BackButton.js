import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faChevronLeft } from '@fortawesome/free-solid-svg-icons';
import { history } from '../routers/AppRouter';

export const BackButton = () => (
  <button
    className="button button--link--dark"
    href="javascript:void(0);"
    onClick={() => history.goBack()}
  >
    <FontAwesomeIcon icon={faChevronLeft} /> Back
  </button>
);
