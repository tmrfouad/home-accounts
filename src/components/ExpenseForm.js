import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';
import { connect } from 'react-redux';

export class ExpenseForm extends React.Component {
  constructor(props) {
    super(props);
    const expense = props.expense;
    this.state = {
      account: expense
        ? expense.account
        : {
            id: '',
            name: ''
          },
      description: expense ? expense.description : '',
      notes: expense ? expense.notes : '',
      amount: expense ? (expense.amount / 100).toString() : '',
      createdAt: expense ? moment(expense.createdAt) : moment(),
      createdAtFocused: false,
      error: '',
      mode: expense ? 'edit' : 'add'
    };
  }

  onAccountChange = e => {
    const id = e.target.value;
    const account = this.props.accounts.find(acc => acc.id === id);
    this.setState(() => ({ account, description: account.name }));
  };

  onDescriptionChange = e => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };

  onNotesChange = e => {
    const notes = e.target.value;
    this.setState(() => ({ notes }));
  };

  onAmountChange = e => {
    const amount = e.target.value;
    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }));
    }
  };

  onCreatedAtChange = createdAt => {
    if (createdAt) {
      this.setState(() => ({ createdAt }));
    }
  };

  onCreatedAtFocusChange = ({ focused: createdAtFocused }) => {
    this.setState(() => ({ createdAtFocused }));
  };

  onSubmit = e => {
    e.preventDefault();
    if (!this.state.account || !this.state.description || !this.state.amount) {
      this.setState(() => ({
        error: 'Please provide account, description and amount.'
      }));
    } else {
      this.setState(() => ({ error: '' }));
      this.props.onSubmit({
        account: this.state.account,
        description: this.state.description,
        amount: parseFloat(this.state.amount) * 100,
        createdAt: this.state.createdAt.valueOf(),
        notes: this.state.notes
      });
    }
  };

  render() {
    return (
      <form className="form" onSubmit={this.onSubmit}>
        {this.state.error && <p className="form__error">{this.state.error}</p>}
        <select
          className="select"
          autoFocus
          value={this.state.account.id}
          onChange={this.onAccountChange}
        >
          <option value="" disabled>
            -- Select Account --
          </option>
          {this.props.accounts.map(acc => (
            <option key={acc.id} value={acc.id}>
              {acc.name}
            </option>
          ))}
        </select>
        <input
          type="text"
          className="text-input"
          placeholder="Description"
          value={this.state.description}
          onChange={this.onDescriptionChange}
        />
        <input
          type="text"
          className="text-input"
          placeholder="Amount"
          value={this.state.amount}
          onChange={this.onAmountChange}
        />
        <SingleDatePicker
          date={this.state.createdAt}
          onDateChange={this.onCreatedAtChange}
          focused={this.state.createdAtFocused}
          onFocusChange={this.onCreatedAtFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
        <textarea
          className="textarea"
          placeholder="Add a note for your expense (optional)"
          value={this.state.notes}
          onChange={this.onNotesChange}
        />
        <div>
          <button className="button">
            {this.state.mode === 'add' ? 'Add Expense' : 'Edit Expense'}
          </button>
        </div>
      </form>
    );
  }
}

const mapStateToProps = state => ({
  accounts: state.accounts
});

export default connect(mapStateToProps)(ExpenseForm);
