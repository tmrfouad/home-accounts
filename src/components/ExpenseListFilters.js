import React from 'react';
import { connect } from 'react-redux';
import {
  setTextFilter,
  sortByDate,
  sortByAmount,
  setStartDate,
  setEndDate
} from '../actions/filters';
import { DateRangePicker } from 'react-dates';

export class ExpenseListFilters extends React.Component {
  state = {
    createdAtfocusedInput: null
  };

  onCreatedAtChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };

  onCreatedAtfocusChange = createdAtfocusedInput => {
    this.setState({ createdAtfocusedInput });
  };

  onTextFilterChange = e => {
    this.props.setTextFilter(e.target.value);
  };

  onSortSelectChange = e => {
    if (e.target.value === 'date') {
      this.props.sortByDate();
    } else if (e.target.value === 'amount') {
      this.props.sortByAmount();
    }
  };

  render() {
    return (
      <div className="content-container">
        <div className="input-group">
          <div className="input-group__item">
            <input
              type="text"
              placeholder="Search expenses"
              className="text-input"
              value={this.props.filters.text}
              onChange={this.onTextFilterChange}
            />
          </div>
          <div className="input-group__item">
            <select
              className="select"
              value={this.props.filters.sortBy}
              onChange={this.onSortSelectChange}
            >
              <option value="date">Date</option>
              <option value="amount">Amount</option>
            </select>
          </div>
          <div className="input-group__item">
            <DateRangePicker
              startDate={this.props.filters.startDate}
              startDateId="your_unique_start_date_id"
              endDate={this.props.filters.endDate}
              endDateId="your_unique_end_date_id"
              onDatesChange={this.onCreatedAtChange}
              focusedInput={this.state.createdAtfocusedInput}
              onFocusChange={this.onCreatedAtfocusChange}
              isOutsideRange={() => false}
              numberOfMonths={1}
              showClearDates={true}
            />
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  filters: state.filters
});

const mapDispatchToProps = dispatch => ({
  setStartDate: startDate => dispatch(setStartDate(startDate)),
  setEndDate: endDate => dispatch(setEndDate(endDate)),
  setTextFilter: text => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ExpenseListFilters);
