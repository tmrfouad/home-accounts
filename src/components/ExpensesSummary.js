import React from 'react';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import { connect } from 'react-redux';
import numeral from 'numeral';
import { Link } from 'react-router-dom';

export const ExpensesSummary = ({
  expenseCount,
  expensesTotal,
  hiddenExpensesCount
}) => (
  <div className="page-header">
    <div className="content-container">
      <h2 className="page-header__title">
        Viewing <span>{expenseCount || 0}</span> expense
        {expenseCount === 1 ? '' : 's'} totalling{' '}
        <span>{numeral(expensesTotal / 100).format('$0,0.00')}</span>
      </h2>
      {hiddenExpensesCount > 0 && (
        <h4>
          Not showing <span>{hiddenExpensesCount || 0}</span> expense
          {hiddenExpensesCount === 1 ? '' : 's'}
        </h4>
      )}
      <div className="page-header__actions">
        <Link className="button" to="/create">
          Add Expense
        </Link>
      </div>
    </div>
  </div>
);

const mapStateToProps = state => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  const hiddenExpensesCount = state.expenses.length - visibleExpenses.length;
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses),
    hiddenExpensesCount: hiddenExpensesCount
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
