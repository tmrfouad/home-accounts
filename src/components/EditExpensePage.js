import React from 'react';
import ExpenseForm from './ExpenseForm';
import ConfirmModal from './ConfirmModal';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import { connect } from 'react-redux';

export class EditExpensePage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isModalOpen: false
    };
  }

  onSubmit = exp => {
    this.props.startEditExpense(this.props.expense.id, exp);
    this.props.history.push('/');
  };

  onRemoveExpense = () => {
    this.props.startRemoveExpense(this.props.expense.id);
    this.props.history.push('/');
  };

  openDialog = () => {
    this.setState(() => ({ isModalOpen: true }));
  };

  closeDialog = () => {
    this.setState(() => ({ isModalOpen: false }));
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h2 className="page-header__title">Edit Expense</h2>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            id="expenseForm"
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
          <button
            className="button button--secondary"
            onClick={this.openDialog}
          >
            Remove Expense
          </button>
        </div>
        <ConfirmModal
          id="confirmModal"
          messageTitle="Remove Expense!"
          messageBody="Are you sure you want to remove this item?"
          isOpen={this.state.isModalOpen}
          onModalClose={this.closeDialog}
          onModalOk={this.onRemoveExpense}
        />
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find(expense => expense.id === props.match.params.id)
});

const mapDispatchToProps = dispatch => ({
  startEditExpense: (id, exp) => {
    dispatch(startEditExpense({ id }, exp));
  },
  startRemoveExpense: id => {
    dispatch(startRemoveExpense({ id }));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditExpensePage);
