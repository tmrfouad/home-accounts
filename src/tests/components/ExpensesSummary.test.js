import React from 'react';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import { shallow } from 'enzyme';

test('should render ExpensesSummary correctly', () => {
  const wrapper = shallow(<ExpensesSummary />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with one expense correctly', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expensesTotal={120} />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with multible expenses correctly', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={5} expensesTotal={840} />
  );
  expect(wrapper).toMatchSnapshot();
});
