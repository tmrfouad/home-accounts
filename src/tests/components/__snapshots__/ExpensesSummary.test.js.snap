// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
      Viewing 
      <span>
        0
      </span>
       expense
      s
       totalling
       
      <span>
        $0.00
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpensesSummary with multible expenses correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
      Viewing 
      <span>
        5
      </span>
       expense
      s
       totalling
       
      <span>
        $8.40
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpensesSummary with one expense correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       expense
       totalling
       
      <span>
        $1.20
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
